test/rebuild.ts

```diff
@@ -52,3 +52,27 @@ test('rebuilds specific dependencies', async function (t: tape.Test) {
   const generatedByPostinstall = project.requireModule('install-scripts-example-for-pnpm/generated-by-postinstall')
   t.ok(typeof generatedByPostinstall === 'function', 'generatedByPostinstall() is available')
 })
+
+test('rebuild with pending option', async function (t: tape.Test) {
+  const project = prepare(t)
+  await installPkgs(['pre-and-postinstall-scripts-example'], testDefaults({ignoreScripts: true}))
+  await installPkgs(['zkochan/install-scripts-example'], testDefaults({ignoreScripts: true}))
+
```

Comment:
you need to verify before calling `rebuild()` that the scripts were not executed, so make the checks that are below but vice versa.

Reply:
Also, check that the ids are added to `.modules.yaml` and are removed from there after running `rebuild()`